From 332f8c052b236b1cc5b0b62aa53582466d495f09 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Mon, 17 Jun 2024 15:01:58 +0200
Subject: [PATCH 0971/1695] dt-bindings: st,stm32-calibration: add STM32MP25
 support

Add support of "st,msi-cal" compatible for STM32MP25

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 64b7ca94762cc3f34055ab02974d149d1c14fff3 dt-bindings: st,stm32-calibration: add STM32MP25 support]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: Id55e426d8d44b191fa95306c632169a2f7a64bd9
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/388016
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 .../bindings/clock/st,stm32-calibration.yaml      | 15 ++++++++++-----
 1 file changed, 10 insertions(+), 5 deletions(-)

diff --git a/documentation/devicetree/bindings/clock/st,stm32-calibration.yaml b/documentation/devicetree/bindings/clock/st,stm32-calibration.yaml
index 89cb5664a..e38958cef 100644
--- a/documentation/devicetree/bindings/clock/st,stm32-calibration.yaml
+++ b/documentation/devicetree/bindings/clock/st,stm32-calibration.yaml
@@ -4,16 +4,20 @@
 $id: http://devicetree.org/schemas/clock/st,stm32-calibration.yaml#
 $schema: http://devicetree.org/meta-schemas/core.yaml#
 
-title: STMicroelectronics STM32 clock calibration bindings
+title: STMicroelectronics STM32 clock calibration
 
 maintainers:
   - Gatien Chevallier <gatien.chevallier@st.com>
 
 description: |
-  STMicroelectronics's STM32 MPUs embed a clock calibration system for HSI
-  and CSI clocks. The calibration relies on a counter. It is a digital
-  calibration circuit with 0.95 ppm resolution, used to compensate quartz
-  crystal inaccuracy.
+  STMicroelectronics's STM32 MPUs embed a clock calibration system for
+  internal oscillators whose frequency can be affected by temperature and
+  voltage variations:
+  * HSI, CSI for STM32MP1 family
+  * HSI, MSI for STM32MP2 family
+  The clock calibration algorithm is based on the comparison of a timer
+  (fed by this oscillator) and a clock that is derived from the HSE clock that
+  is considered always accurate.
 
 properties:
   compatible:
@@ -27,6 +31,7 @@ patternProperties:
         enum:
           - "st,csi-cal"
           - "st,hsi-cal"
+          - "st,msi-cal"
       counter:
         description: counter used to calibrate
           Refer to counter.yaml for more details.
-- 
2.47.2

